src/game/OperationHandling.hpp

```diff
@@ -10,47 +10,93 @@
         void operator()(Event &&, FSM &fsm, SourceState &, TargetState &) {
+            spdlog::info("Broadcasting state");
+            spy::gameplay::State &s = root_machine(fsm).gameState;
+            MessageRouter &router = root_machine(fsm).router;
 
-            fsm.activeCharacter = fsm.remainingCharacters.front();
-            fsm.remainingCharacters.pop_front();
-
-            spdlog::info("Requesting GameOperation for character {}", fsm.activeCharacter);
-            // TODO: find player owning activeCharacter
-            auto activePlayer = Player::one;
-            spdlog::info("Requesting GameOperation from player {}", activePlayer);
-            // TODO properly request Operation:
-            //  root_machine(fsm).router.sendMessage(static_cast<int>(activePlayer), spy::network::messages::RequestGameOperation());
+            spy::network::messages::GameStatus g{{}, fsm.activeCharacter, fsm.operations, s, false};
+            router.broadcastMessage(g);
+            fsm.operations.clear();
         }
     };
 
     /**
-     * @brief calls both actions::handleOperation and actions::requestNextOperation with the same parameters
+     * Request operation from next character in list
      */
-    struct handleOperationAndRequestNext {
+    struct requestNextOperation {
         template<typename Event, typename FSM, typename SourceState, typename TargetState>
-        void operator()(Event &&event, FSM &fsm, SourceState &source, TargetState &target) {
-            // TODO find a way to directly specify multiple actions in transition table
-            handleOperation{}(event, fsm, source, target);
-            requestNextOperation{}(event, fsm, source, target);
+        void operator()(Event &&, FSM &fsm, SourceState &, TargetState &) {
+            spdlog::trace("requestNextOperation for character {}", fsm.activeCharacter);
+            spy::gameplay::State &state = root_machine(fsm).gameState;
+
+            // Find character object by UUID to determine faction
+            auto activeChar = std::find_if(state.getCharacters().begin(),
+                                           state.getCharacters().end(),
+                                           [&fsm](const spy::character::Character &c) {
+                                               return c.getCharacterId() == fsm.activeCharacter;
+                                           });
+
+            Player activePlayer;
+            switch (activeChar->getFaction()) {
+                case spy::character::FactionEnum::PLAYER1:
+                    activePlayer = Player::one;
+                    break;
+                case spy::character::FactionEnum::PLAYER2:
+                    activePlayer = Player::two;
+                    break;
+                default:
+                    spdlog::critical("activeCharacter ({}) does not have valid faction", fsm.activeCharacter);
+                    std::exit(1);
```

Comment:
Fallen nicht alle NPCs in diesen Fall?

Reply:
FactionEnum hat bisher keinen wert für NPCs, bei den meta informationen im std gibt es `Faction.Neutral`. Ich mache das zu einer warning, da wir das noch nicht implementiert haben.